tests/integrations/raftstore/test_flashback.rs

```diff
@@ -0,0 +1,273 @@
+    block_on(cluster.call_and_wait_prepare_flashback(region.get_id(), 1));
+
+    // write will be blocked
+    let value = vec![1_u8; 8096];
+    must_get_error_recovery_in_progress(&mut cluster, &region, new_put_cmd(b"k1", &value));
+
+    must_cmd_add_flashback_flag(
+        &mut cluster,
+        &mut region.clone(),
+        new_put_cmd(b"k1", &value),
+    );
+
+    cluster.call_finish_flashback(region.get_id(), 1);
+
+    multi_do_cmd(&mut cluster, new_put_cf_cmd("write", b"k1", &value));
+}
+
+#[test]
+fn test_flahsback_for_read() {
+    let mut cluster = new_node_cluster(0, 3);
+    cluster.run();
+
+    // write for cluster
+    let value = vec![1_u8; 8096];
+    multi_do_cmd(&mut cluster, new_put_cf_cmd("write", b"k1", &value));
+    // read for cluster
+    multi_do_cmd(&mut cluster, new_get_cf_cmd("write", b"k1"));
+
+    // prepare for flashback
+    let region = cluster.get_region(b"k1");
+    block_on(cluster.call_and_wait_prepare_flashback(region.get_id(), 1));
+
+    // read will be blocked
+    must_get_error_recovery_in_progress(&mut cluster, &region, new_get_cf_cmd("write", b"k1"));
+
+    // verify the read can be executed if add flashback flag in request's
+    // header.
+    must_cmd_add_flashback_flag(
+        &mut cluster,
+        &mut region.clone(),
+        new_get_cf_cmd("write", b"k1"),
+    );
+
+    cluster.call_finish_flashback(region.get_id(), 1);
+
+    multi_do_cmd(&mut cluster, new_get_cf_cmd("write", b"k1"));
+}
+
+// LocalReader will try to renew lease in advance,
+// but when enable flashback, will make lease at None and prevent renew lease.
+#[test]
+fn test_flahsback_for_local_read() {
+    let mut cluster = new_node_cluster(0, 3);
+    let election_timeout = configure_for_lease_read(&mut cluster, Some(50), None);
+
+    // Avoid triggering the log compaction in this test case.
+    cluster.cfg.raft_store.raft_log_gc_threshold = 100;
+
+    let node_id = 3u64;
+    let store_id = 3u64;
+    let peer = new_peer(store_id, node_id);
+    cluster.run();
+
+    cluster.must_put(b"k1", b"v1");
+    let region = cluster.get_region(b"k1");
+    cluster.must_transfer_leader(region.get_id(), peer.clone());
+
+    // check local read before prepare flashback
+    let state = cluster.raft_local_state(region.get_id(), store_id);
+    let last_index = state.get_last_index();
+    // Make sure the leader transfer procedure timeouts.
+    std::thread::sleep(election_timeout * 2);
+    must_read_on_peer(&mut cluster, peer.clone(), region.clone(), b"k1", b"v1");
+    // Check the leader does a local read.
+    let state = cluster.raft_local_state(region.get_id(), store_id);
+    assert_eq!(state.get_last_index(), last_index);
+
+    // prepare for flashback
+    block_on(cluster.call_and_wait_prepare_flashback(region.get_id(), store_id));
+
+    must_error_read_on_peer(
+        &mut cluster,
+        peer.clone(),
+        region.clone(),
+        b"k1",
+        Duration::from_secs(1),
+    );
+
```

Comment:
What about checking the `last_index` here also to make sure the read is blocked totally?